### PR TITLE
Move "utils" bindings to cxx crate

### PR DESCRIPTION
**Pull request by Minoru**
Just a bit of churn. Some functions turned out to be harder to move, so I split those into separate commits to make the review easier.

Reviews are welcome! I'll merge this in three days if no issues are raised.

### CODE REVIEW

File: rust/libnewsboat/src/utils.rs [hunk 1]
```diff
@@ -858,21 +858,33 @@ pub fn strip_comments(line: &str) -> &str {
...
+pub fn extract_filter(line: &str) -> FilterUrlParts {
     debug_assert!(line.starts_with("filter:"));
     // line must start with "filter:"
     let line = line.get("filter:".len()..).unwrap();
-    let (filter, url) = line.split_at(line.find(':').unwrap_or(0));
+    let (script_name, url) = line.split_at(line.find(':').unwrap_or(0));
     let url = url.get(1..).unwrap_or("");
     log!(
         Level::Debug,
         "utils::extract_filter: {} -> filter: {} url: {}",
```
Comment (der-lyse) [inline rust/libnewsboat/src/utils.rs:879]: It's just a debug log, but for consistency the "filter" could be renamed to "script name", too.

**Review by dennisschagt — APPROVED** [review]
Nice improvements and reduction in FFI-related lines-of-code.
LGTM

**Review by dennisschagt — APPROVED** [review]
:+1:

**Review by der-lyse — APPROVED** [review]
Nice!

**Comment by Minoru** [comment]
Thanks for the reviews! I squashed the changes and will merge once the CI is done.

**Pull request merged**